specs/corla.pvs

```diff
@@ -0,0 +1,1669 @@
+
+%+BEGIN elections
+
+%%% \section{Elections}
+
+%%% Elections have a number of general concepts necessary to specify
+%%% the RLA Tool. Some are generic to all elections, and others are
+%%% specifiec to Colorado.
+
+elections: THEORY
+BEGIN
+%%% Voters are the most important concept in elections, as they are
+%%% who elections are for, and who should make the choices that
+%%% determine the outcome of legitimate elections. Some voters are
+%%% UOCAVA voters. What follows is one way to formalize such, simply
+%%% stating that the type uocava_voter is a predicate on the type
+%%% voter. Consequently, all voters are UOCAVA or not.
+  person: TYPE
+  voter: TYPE FROM person
+  elector: TYPE = voter
+  uocava_voter: TYPE from voter
+
+%%% The dual of voters are candiates, who run for office to represent
+%%% voters. Note that a person can be either a voter, a candidate,
+%%% both, or neither.
+  candidate: TYPE FROM person
+
+%%% Elections focus on contests, each of which represents a set of
+%%% choices that voters can make. Making a legal choice ranges
+%%% enormously across the Earth, from marking a vote for a candidate
+%%% by filling in a single bubble to enumerating a total order on all
+%%% choices in a contest. Across a given ballot, each contest has a
+%%% unique name.
+% @review CDOS will there be one risk limit? If more than one, then
+% system must provide a way to map contests to risk limits.
+
+  contest: TYPE
+  contest_name?: [contest -> string]
+% @trace RDPR-6-Jul-2017 Section 25.1.2
+% @description A contest selected by the Secretary of State for a
+% risk-limiting audit.
+  audited_contest: TYPE FROM contest
+  opportunistic_contest: TYPE FROM contest
+  full_hand_count_contest: TYPE FROM contest
+% @todo Contest name uniqueness.
+
+  outcome: TYPE
+  election_outcome: TYPE = set[outcome]
+  wrong_outcome: TYPE FROM election_outcome
+  
+% @trace RDPR-6-Jul-2017 Section 25.1.5
+% @description The presumed winning and losing candidates or voting
+% choices of a ballot contest as reflected in preliminary results.
+% @todo kiniry TBD
+  reported_tabulation_outcome: TYPE FROM election_outcome
+
+% @trace RDPR-6-Jul-2017 Section 25.2.2(D)
+% @explanation On the ninth day after Election Day, the county must
+% finish tabulating all ballots other than provisional ballots and
+% property owner ballots.
+  rla_tabulation_outcome: TYPE FROM election_outcome
+
+%%% Elections are defined across cohorts of voters arranged in any
+%%% number of ways---geographic, political, professional, and more. In
+%%% Colorado, elections are organized across counties, and the entire
+%%% state. Each county has a name and a county number, both of which
+%%% are unique across the whole state.
+
+  county: TYPE
+  county_name?: [county -> string]
+  county_number?: [county -> nat]
+  state: TYPE = set[county]
+  nation: TYPE = set[state]
+  organization: TYPE
+
+  s: VAR state
+  c1, c2: VAR county
+
+  county_names_unique: AXIOM
```

Comment:
This axiom and the following one have a state `s` as a bound variable in the quantifier that is unused; it should explicitly say that `c1` and `c2` are both members of `s` (that is, both counties must be in the same state for uniqueness to be required of their names and numbers).